Modules/Scripted/Endoscopy/Endoscopy.py

```diff
@@ -834,11 +722,186 @@ def interpolateOrientations(self):
 
-    def getDefaultOrientation(self, curve, curveControlPointIndex):
+    def saveOrientationAtIndex(self, resampledCurvePointIndex):
+        # Compute CameraOrientations dictionary key
+        distanceAlongResampledCurve = EndoscopyLogic.distanceAlongCurveOfNthControlPoint(
+            self.resampledCurve, resampledCurvePointIndex,
+        )
+
+        # Compute CameraOrientations dictionary value
+        cameraNode = EndoscopyLogic.getCameraFromInputCurve(self.inputCurve)
+        cameraPosition = cameraNode.GetPosition()
+        focalPoint = cameraNode.GetFocalPoint()
+        viewUp = cameraNode.GetViewUp()
+        worldMatrix3x3 = EndoscopyLogic.buildCameraMatrix3x3(cameraPosition, focalPoint, viewUp)
+        worldOrientation = EndoscopyLogic.matrix3x3ToOrientation(worldMatrix3x3)
+
+        # Add new dictionary key-value pair
+        cameraOrientations = EndoscopyLogic.getCameraOrientationsFromInputCurve(self.inputCurve)
+        cameraOrientations[distanceAlongResampledCurve] = worldOrientation
+
+        EndoscopyLogic.setInputCurveCameraOrientations(self.inputCurve, cameraOrientations)
+        self.interpolateOrientationsForControlPoints(cameraOrientations)
+
+        return cameraOrientations
+
+    def removeOrientationAtIndex(self, resampledCurvePointIndexToDelete):
+        cameraOrientations = EndoscopyLogic.getCameraOrientationsFromInputCurve(self.inputCurve)
+
+        # The inputCurve (and hence the resampledCurve) could have been modified since these distances were saved,
+        # meaning that computing a distance from an index may not now give the same distance value.  However, we can
+        # delete the entries associated with this resampledCurvePointIndexToDelete even in the case that its
+        # distance has changed.
+
+        # We must freeze the cameraOrientations.keys() generator at the start (by converting it to a list) because we
+        # delete from the cameraOrientations Python dict in this loop.
+        for distanceAlongResampledCurve in list(cameraOrientations.keys()):
+            resampledCurvePointIndex = EndoscopyLogic.indexOfControlPointForDistanceAlongCurve(
+                self.resampledCurve, distanceAlongResampledCurve,
+            )
+            if resampledCurvePointIndex == resampledCurvePointIndexToDelete:
+                del cameraOrientations[distanceAlongResampledCurve]
+                break
+
+        EndoscopyLogic.setInputCurveCameraOrientations(self.inputCurve, cameraOrientations)
+        self.interpolateOrientationsForControlPoints(cameraOrientations)
+
+        return cameraOrientations
+
+    def updateCameraFromOrientationAtIndex(self, resampledCurvePointIndex):
+        """Apply the resampledCurvePointIndex-th step in the path to the camera"""
+
+        if (
+            self.resampledCurve is None
+            or not 0 <= resampledCurvePointIndex < self.getNumberOfControlPoints()
+        ):
```

Comment:
These four lines could be collapsed into one line
```python
if self.resampledCurve is None or not 0 <= resampledCurvePointIndex < self.getNumberOfControlPoints():
```

Reply:
This was preemptively formatted to comply with `ruff format` (that we will soon fully support in Slicer)